hints/darwin.sh

```diff
@@ -364,6 +364,10 @@ esac
 # Darwin's querylocale() has races
 ccflags="$ccflags -DNO_THREAD_SAFE_QUERYLOCALE"
 
+# But it doesn't much matter because the whole implementation has bugs [GH
+# #21556]
+ccflags="$ccflags -DNO_POSIX_2008_LOCALE"
+
```

Comment:
The patch itself is minimal, therefore easily rolled back if harmful.  So +1 from me -- but I have no Darwin to test this on.